moebot_bot/bot/commands/permit.go

```diff
@@ -11,18 +11,42 @@ type PermitCommand struct {
 }
 
 func (pc *PermitCommand) Execute(pack *CommPackage) {
-	// should always have more than 2 params: permission level, role name ... role name
-	if len(pack.params) < 2 {
-		pack.session.ChannelMessageSend(pack.message.ChannelID, "Please provide a permission level followed by the role name")
+	args := ParseCommand(strings.Join(pack.params, " "), []string{"-permission", "-securityAnswer", "-confirmationMessage"})
+
+	if _, ok := args[""]; !ok || len(args) <= 1 {
+		pack.session.ChannelMessageSend(pack.message.ChannelID, "Please provide a role name followed by one or more arguments.")
 		return
 	}
-	permLevel := db.GetPermissionFromString(pack.params[0])
-	if permLevel == -1 {
-		pack.session.ChannelMessageSend(pack.message.ChannelID, "Invalid permission level")
-		return
+
+	role := db.Role{}
+
+	if perm, ok := args["-permission"]; ok {
+		permLevel := db.GetPermissionFromString(perm)
+		if permLevel == -1 {
+			pack.session.ChannelMessageSend(pack.message.ChannelID, "Invalid permission level")
+			return
+		}
+		role.Permission = permLevel
 	}
+
+	if sa, ok := args["-securityAnswer"]; ok {
+		if sa == "" {
+			pack.session.ChannelMessageSend(pack.message.ChannelID, "Please provide a valid security answer")
+			return
+		}
+		role.ConfirmationSecurityAnswer.Scan(sa)
```

Comment:
Ohhh that's actually a cleaner way than I was doing it in the past (setting valid + string manually) 👍